stdlib/src/collections/optional.mojo

```diff
@@ -100,7 +100,11 @@ struct Optional[T: CollectionElement](CollectionElement, Boolable):
         self = Self()
 
     @always_inline
-    fn value(self) -> T:
+    fn value[
+        mutability: __mlir_type.`i1`, self_life: AnyLifetime[mutability].type
+    ](self: Reference[Self, mutability, self_life].mlir_ref_type) -> Reference[
+        T, mutability, self_life
+    ]:
         """Unsafely retrieve the value out of the Optional.
 
         This function currently creates a copy. Once we have lifetimes
```

Comment:
**Suggestion** Please update the doc strings here as we no longer create a copy 😄